### PR TITLE
Various improvements for Rings-For-All

### PR DESCRIPTION
**Pull request by lL1l1**

### CODE REVIEW

File: mods/RFA/hook/lua/ui/controls/worldview.lua [hunk 1]
```diff
@@ -238,26 +356,44 @@ do
...
             local ring = Ring()
-            ring.pos = GetMouseWorldPos()
-            ring.radius = (radius or 0) + 2
+            if useMousePos then
+                ring.pos = GetMouseWorldPos()
+            else
+                local unit = builders[1]
+                local pos = unit:GetPosition()
+                local queue = unit:GetCommandQueue()
+                for i = table.getn(queue), 0, -1 do
```
Comment (4z0t) [inline mods/RFA/hook/lua/ui/controls/worldview.lua:386]: Is this list 0-indexed?

File: mods/RFA/hook/lua/ui/controls/worldview.lua [hunk 2]
```diff
@@ -238,26 +356,44 @@ do
...
-            local engineers = EntityCategoryFilterDown(categories.ENGINEER, selection)
-            if TableEmpty(engineers) then
+            ---@type UserUnit[]
+            local builders = EntityCategoryFilterDown(buildersCategory, selection)
+            if TableEmpty(builders) then
                 self:ClearBuildRings()
                 return
             end
 
             local radius = selection
```
Comment (4z0t) [inline mods/RFA/hook/lua/ui/controls/worldview.lua:374]: If you use `builders` in picking position, should you pick radius for `builders` too?

**Review by 4z0t — CHANGES_REQUESTED** [review]
Not final one, only looked through the code.

**Comment by 4z0t** [comment]
I'd also update UMT for some new LuaQ functions, but it is for later.

File: mods/RFA/hook/lua/ui/controls/worldview.lua (hunk 2 above)
Comment (lL1l1) [reply mods/RFA/hook/lua/ui/controls/worldview.lua]: The result is the same but the logic is somewhat different imo: you shouldn't check build radius for units that can't build

**Pull request merged**